core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -56,18 +56,24 @@ class CloudSqlInstance {
   private final CloudSqlInstanceName instanceName;
   private final ListenableFuture<KeyPair> keyPair;
   private final Object instanceDataGuard = new Object();
-  // Limit forced refreshes to 1 every minute.
-  private final RateLimiter<Object> forcedRenewRateLimiter =
-      RateLimiter.burstyBuilder(2, Duration.ofSeconds(30)).build();
-
+  // Rate limit delay calculator
+  private final RateLimitCalculator rateLimitCalculator =
+      new RateLimitCalculator(
+          Duration.ofMillis(50), Duration.ofMillis(250), 2.0f, Duration.ofMinutes(2));
   private final RefreshCalculator refreshCalculator = new RefreshCalculator();
 
   @GuardedBy("instanceDataGuard")
   private ListenableFuture<InstanceData> currentInstanceData;
 
+  @GuardedBy("instanceDataGuard")
```

Comment:
New future for the initialInstanceData is used to get the first successful refresh attempt. It's returned when getInstanceData() is called before peformRefresh() succeeds.